ligo/stablecoin/spec.ligo

```diff
@@ -0,0 +1,263 @@
+
+type owner_transfer_policy_ is
+| Owner_no_op
+| Optional_owner_hook
+| Required_owner_hook
+
+type owner_transfer_policy is michelson_or_right_comb(owner_transfer_policy_)
+
+type custom_permission_policy_ is record
+  tag        : string
+; config_api : option (address)
+end
+
+type custom_permission_policy is michelson_pair_right_comb(custom_permission_policy_)
+
+type permissions_descriptor_ is record
+  self     : self_transfer_policy
+; operator : operator_transfer_policy
+; receiver : owner_transfer_policy
+; sender   : owner_transfer_policy
+; custom   : option (custom_permission_policy)
+end
+
+type permissions_descriptor is michelson_pair_right_comb(permissions_descriptor_)
+
+type permissions_descriptor_params is
+  contract (permissions_descriptor)
+
+type mint_params is record
+  to_    : address
+; amount : nat
+end
+
+type burn_params is record
+  from_  : address
+; amount : nat
+end
+
+type parameter is
+  Transfer               of transfer_params
+| Balance_of             of balance_of_params
+| Total_supply           of total_supply_params
+| Token_metadata         of token_metadata_params
+| Permissions_descriptor of permissions_descriptor_params
+| Update_operators       of update_operator_params
+| Is_operator            of is_operator_params
+
+(* ------------------------------------------------------------- *)
+
+(*
+ * Hooks
+ *)
+
+type registry is set (address)
+
+// TODO: currently it's the same as `transfer` but with optional
+// participants.
+type transfer_descriptor_ is record
+  from_    : option (address)
+; to_      : option (address)
+; token_id : token_id
+; amount   : nat
+end
+
+type transfer_descriptor is michelson_pair_right_comb(transfer_descriptor_)
+
+type transfer_descriptor_param_ is record
+   fa2      : address
+;  batch    : list(transfer_descriptor)
+;  operator : address
+end
+
+type transfer_descriptor_param is michelson_pair_right_comb(transfer_descriptor_param_)
+
+type owner_hook_ is
+| Tokens_received of transfer_descriptor_param
+| Tokens_sent of transfer_descriptor_param
+
+type owner_hook is michelson_or_right_comb(owner_hook_)
+
+type transfer_hook is unit -> contract (transfer_descriptor_param)
+
+type set_hook_param_ is record
+  hook : transfer_hook
+; permissions_descriptor : permissions_descriptor // TODO: unused
+end
+
+type set_hook_param is michelson_pair_right_comb (set_hook_param_)
+
+(* ------------------------------------------------------------- *)
+
+(* Stablecoin parameter *)
+type closed_parameter is
+| Call_FA2 of parameter
+| Set_transfer_hook of set_hook_param
+
+(* ------------------------------------------------------------- *)
+
+(*
+ * Storage
+ *)
+
+type operator_actions is
+| All_operator_tokens
+| Some_operator_tokens of set (token_id)
+| All_operator_tokens_except of set (token_id)
+
+type owner is address
+type operator is address
+
+type operators is
+  big_map ((owner * operator), unit)
+
+// TODO: currently we need to correct our haskell tests
```

Comment:
I am not sure what it means, but do we still need it? Looks like a leftover. `paused` field is out of scope of this issue, but we can have it because we'll need it later. The thing I dislike is that:
1. `storage_skeleton` is called this way. Its record name is `fields`, so why not `storage_fields` then?
2. If it's supposed to have all fields beside big_maps, why doesn't it have `total_supply` and metadata?

That being said, I would just remove it.